data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -154,13 +166,60 @@ private long calculateLongestThreadWaitingTime() {
                 orElse(1L);
     }
 
-    private Properties getConsumerProperties(TopicConfig topicConfig) {
+    public String getBootStrapServersForMsk(final AwsIamAuthConfig awsIamAuthConfig, final AwsConfig awsConfig) {
+        ClientConfiguration clientConfiguration = new ClientConfiguration();
+        AWSCredentialsProvider credentialProvider;
+        if (awsIamAuthConfig == AwsIamAuthConfig.DEFAULT) {
+            credentialProvider = new DefaultAWSCredentialsProviderChain();
+        } else if (awsIamAuthConfig == AwsIamAuthConfig.ROLE) {
+            String sessionName = "data-prepper-kafka-session"+UUID.randomUUID();
+            credentialProvider = new STSAssumeRoleSessionCredentialsProvider.Builder(awsConfig.getStsRoleArn(), sessionName).build();
+        } else {
+            throw new RuntimeException("Unknown AWS IAM auth mode");
+        }
+        AWSKafka kafkaClient = AWSKafkaClientBuilder.standard()
+                .withClientConfiguration(clientConfiguration)
+                .withCredentials(credentialProvider)
+                .withRegion(awsConfig.getRegion())
+                .build();
+        GetBootstrapBrokersRequest request = new GetBootstrapBrokersRequest();
+        String clusterArn = awsConfig.getAwsMskConfig().getArn();
+        request.setClusterArn(clusterArn);
+        GetBootstrapBrokersResult result = kafkaClient.getBootstrapBrokers( request );
+        // TODO return this based on the broker_connection_mode
+        return result.getBootstrapBrokerStringSaslIam();
```

Comment:
This should be bootstrapBrokerStringVpcConnectivitySaslIam()
Also would be good to add try/catch and log exception

Reply:
For some reason bootstrapBrokerStringVpcConnectivitySaslIam() is not working. I added "TODO" comment to investigate further and fix it later.

Reply:
I am getting the error `symbol not found` when using that API. I will investigate and update it in the next PR.

Reply:
I may have to move to newer version of SDK. Will do it in the next PR.